### PR TITLE
[MM-28064] Add telemetry in various places around cloud message banners

### PR DESCRIPTION
**Pull request by nickmisasi**
#### Summary
Adds telemetry according to: https://docs.google.com/document/d/102uElsng_cnpFHAr8tqTEIhktX5x5R9LAxDLBPDOvAU/edit#heading=h.tgaxd6frqkfr

Note that not all banners in the list exist. This PR includes 
```
User presented with light blue user limit contextual banner ‘Just a heads up … you’re nearing the user limit with the free tier of Mattermost Cloud. We’ll let you know when you hit the limit’
```
and lower in the table. 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-28064

#### Related Pull Requests
n/a
#### Screenshots
n/a

### CODE REVIEW

**Review by Adovenmuehle — APPROVED** [review]
:shipit:

**Review by ethervoid — APPROVED** [review]
LGTM!

File: components/announcement_bar/cloud_announcement_bar/user_limit_announcement_bar.tsx [hunk 1]
```diff
@@ -107,23 +139,30 @@ export default class UserLimitAnnouncementBar extends React.PureComponent<Props>
...
-        }
+        const dismissable = this.isDismissable();
 
         return (
             <AnnouncementBar
                 type={dismissable ? AnnouncementBarTypes.ADVISOR : AnnouncementBarTypes.CRITICAL_LIGHT}
                 showCloseButton={dismissable}
                 handleClose={this.handleClose}
-                showModal={() =>
+                showModal={() => {
```
Comment (devinbinnie) [inline components/announcement_bar/cloud_announcement_bar/user_limit_announcement_bar.tsx:149]: Nit: This is kinda too big to be in-line, would prefer it as a external function.

**Review by devinbinnie — APPROVED** [review]

File: components/announcement_bar/cloud_announcement_bar/user_limit_announcement_bar.tsx (hunk 1 above)
Comment (nickmisasi) [reply components/announcement_bar/cloud_announcement_bar/user_limit_announcement_bar.tsx]: Moved this body into proper function

**Pull request merged**